### PR TITLE
feat: add test helper

### PR DESCRIPTION
**Pull request by Eomm**
closes https://github.com/fastify/fastify-cli/issues/416

### CODE REVIEW

**Review by mcollina — COMMENTED** [review]
Can you integrate this in the generated scaffold?

**Comment by Eomm** [comment]
@fastify/typescript 
I can't find a way to let this PR work.

```
templates/app-ts/test/routes/example.test.ts 2> TSError: ⨯ Unable to compile TypeScript:
templates/app-ts/test/routes/example.test.ts 2> templates/app-ts/test/helper.ts(3,23): error TS7016: Could not find a declaration file
for module 'fastify-cli/helper'. '/Users/mspigolon/workspace/fastify-cli/node_modules/fastify-cli/helper.js' implicitly has an 'any'
type.
```

Should I add the typings to this project?

**Comment by Eomm** [comment]
@mcollina the PR is ready.

It does not update the TS template. I would open an issue for it, or I need some help from @fastify/typescript to make it works

File: templates/app/test/routes/root.test.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 'use strict'
 
 const { test } = require('tap')
-const { build } = require('../helper')
+const { buildApplication } = require('../helper')
 
 test('default root route', async (t) => {
-  const app = build(t)
+  const app = await buildApplication(t)
```
Comment (mcollina) [inline templates/app/test/routes/root.test.js:7]: why do you need to await this? It's not strictly needed by fastify.

File: README.md [hunk 2]
```diff
@@ -323,6 +323,41 @@ The documentation is rendered using an interactive terminal session that you can
 
 run `fastify docs` to get started.
 
+## Test helpers
+
```
Comment (mcollina) [inline README.md:327]: make it clear that the test helper are only for tap and not jest/mocha/tape etc.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: templates/app/test/routes/root.test.js (hunk 1 above)
Comment (Eomm) [reply templates/app/test/routes/root.test.js:7]: It is not required by fastify, but by the fastify-cli logic.

for example we hit this line:

https://github.com/fastify/fastify-cli/blob/5654ea1710c5b8e67dd5e73bcfc5349a983bdb10/start.js#L89

that read the package.json and manage the ESM files.

It should not be necessary if we manage simple CJS script

File: README.md [hunk 3]
```diff
@@ -323,6 +323,41 @@ The documentation is rendered using an interactive terminal session that you can
...
+To do so, you can use the this module to load your application and give you the control to write your assertions.
+
+There are two utilities provided:
+
+- `build`: builds your application and returns the `fastify` instance without calling the `listen` method.
+- `listen`: starts your application and returns the `fastify` instance listening on the configured port.
+
+Both of these utilities have the `function(arg, pluginOptions)` parameters:
+
+- `arg`: is a string or a string array within the same arguments passed to the `fastify-cli` command.
```
Comment (RafaelGSS) [inline README.md:338]: I would rename it to something more understandable, what o you think about `cliArgs` ?

**Review by RafaelGSS — APPROVED** [review]
LGTM.

File: templates/app/test/helper.js [hunk 4]
```diff
@@ -14,13 +14,14 @@ function config () {
 }
 
 // automatically build and tear down our instance
-function build (t) {
-  const app = Fastify()
+async function buildApplication (t) {
```
Comment (mcollina) [inline templates/app/test/helper.js:17]: let's keep calling this `build`.

**Review by mcollina — CHANGES_REQUESTED** [review]
nits, good work

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**